src/analyz/to_json.rs

```diff
@@ -536,11 +559,16 @@ impl ToJson<'_> for Mutability {
     }
 }
 
-impl ToJson<'_> for NullOp {
+impl<'tcx> ToJson<'tcx> for NullOp<'tcx> {
     fn to_json(&self, _: &mut MirState) -> serde_json::Value {
         match self {
             NullOp::SizeOf => json!({ "kind": "SizeOf" }),
             NullOp::AlignOf => json!({ "kind": "AlignOf" }),
+            NullOp::UbChecks => json!({ "kind": "UbChecks" }),
+            NullOp::ContractChecks => json!({ "kind": "ContractChecks" }),
+
+            // Data-carrying variants — use Debug formatting
+            NullOp::OffsetOf { .. } => json!({ "kind": format!("{:?}", self) }),
```

Comment:
Upon further thought, I'm not quite sure if this is right. The field of `OffsetOf` has type `&List<(VariantIdx, FieldIdx)>`, which we likely want to serialize to JSON rather than relying on its `Debug` output.